### DISCUSSION TITLE
Discriminated Union

### DISCUSSION THREAD

**Discussion by o-az** [open-circle/valibot#148] [Q&A] [answered]
Is there an API for union? I want a check for either property X or property Y exists in an object. In Zod I used https://zod.dev/?id=discriminated-unions [Discriminated unions](https://zod.dev/?id=discriminated-unions)

Example: parsing environment variables and checking either `NEXT_PUBLIC_SITE_URL` or `NEXT_PUBLIC_VERCEL_URL` exists.

**Comment by fabian-hiller** [comment]
For now, just use the `union` schema function of Valibot. Der difference to `discriminatedUnion` is only performance. Zod plans to replace `discriminatedUnion` with something like `switch` in the future: https://github.com/colinhacks/zod/issues/2106

**Comment by CanRau** [comment, marked as answer]
Available since v0.20.0 https://github.com/fabian-hiller/valibot/issues/90
